### ISSUE TITLE
Notes: Use the correct accessibility tree

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#72594]
## What problem does this address?

In the current Notes sidebar, each note has the `listitem` role and is wrapped in an element with the `list` role. It's composed of the following accessibility tree:

```
- list
  - listitem: "Note: This is a first note."
    - button: "Add new note"
    - image: "User avatar"
    - StaticText: "Admin"
    - time
    - button: "Resolve"
    - button: "Actions"
    - paragraph > "StaticText: This is a first note."
    - image: "User avatar"
    - StaticText: "Admin"
    - time
    - paragraph > "StaticText: This is a first reply."
    - image: "User avatar"
    - StaticText: "Admin"
    - time
    - paragraph > "StaticText: This is the second reply."
    - button: "Back to block"
  - listitem: "Note: This is a first note."
    - ... 
```

I think there are at least two problems with the current implementation:

- The `listitem` element is expandable. As [this comment](https://github.com/WordPress/gutenberg/pull/72103#discussion_r2405983571) indicates, the `listitem` role may be inappropriate for expandable elements.
- Because the elements in the list items are all parallel, it's hard to tell which are first notes and which are replies.

## What is your proposed solution?

As far as I can tell, we might be able to use `tree` and  `treeitem` for expandable nested elements.

I think a tree like this might be best, but I'm looking forward to feedback.

```
- tree
  - treeitem: "Note: This is a first note."
    - button: "Add new note"
    - image: "User avatar"
    - StaticText: "Admin"
    - time
    - button: "Resolve"
    - button: "Actions"
    - paragraph > "StaticText: This is a first note."
    - group
      - treeitem
        - image: "User avatar"
        - StaticText: "Admin"
        - time
        - paragraph > "StaticText: This is a first reply."
      - treeitem
        - image: "User avatar"
        - StaticText: "Admin"
        - time
        - paragraph > "StaticText: This is the second reply."
    - button: "Back to block"
  - treeitem: "Note: This is a first note."
    - ... 
```

cc @joedolson @alexstine @afercia

**Comment by talldan** [comment]
> As far as I can tell, we might be able to use tree and treeitem for expandable nested elements.

TreeGrid, (like ListView uses) is also another option, as I think the regular Tree might not support interactive elements inside it (IIRC it should only have 1-dimensional keyboard nav).

Screenreader support is usually not great for these kinds of more complex roles though, so some parts need to be improvised.

**Comment by joedolson** [comment]
I did find the flat hierarchy to be concerning. Using a structure that provided a relationship between a note and its replies would be an improvement, for sure. And it is true that `aria-expanded` is not permitted on `listitem`, which is not expected as a control, only as a structure. `treeitem` would work.

It would be nice if objects could have multiple roles, but I guess that's a problem for future specs...

**Comment by t-hamano** [comment]
Thank you everyone for your feedback! I'd like to test the improvements and submit a pull request.

**Comment by t-hamano** [comment]
The Beta3 release is coming soon, and to allow for further discussion on the correct approach, I'm punting this PR to 6.9.1. If it doesn't make it into the minor release either, I'll punt it to 7.0.

**Comment by adamsilverstein** [comment]
> The Beta3 release is coming soon, and to allow for further discussion on the correct approach, I'm punting this PR to 6.9.1. If it doesn't make it into the minor release either, I'll punt it to 7.0.

@joedolson can you confirm the approach outlined above is the best path forward?

**Comment by joedolson** [comment]
Yes, this is better than what's currently there. As it stands right now, the use of the listitem to add clickable behavior isn't totally reliable; not all browser/AT combinations will expand and collapse the comments when a screen reader is running.

The tree needs some polish:

Important for initial implementation:

1 The container with `role="tree"` must have an `aria-label`, so each can be distinguished. Probably named by the first comment text; preferably using `aria-labelledby` pointing to that content.
2. The first comment should be a treeitem with children, with aria-expanded="true|false".
3. Successive replies should be treeitems that are children of the first comment.

Technically, the tree and the expandable tree item should be separately expandable. But I think that this is potentially good for the future; it would open the door for multiple independent note streams on a single block. Right now, a block can only have one note thread - using a tree means that the note thread can have multiple threads. 

Needs to be done, could be a point release:

4. To add the necessary tree navigation, the right arrow should open the tree item (same as Enter currently does), and left arrow should close it when the parent node is focused. Up/Down arrows should navigate between individual tree items. Home should go to the first comment; End should go to the last comment.

**Comment by joedolson** [comment]
The change to give the top level tree an accessible name, however, is something that really should happen in 6.9. The other structural things could possibly wait; they aren't critical. The missing accessible name is a pretty clear bug.

**Comment by jeffpaul** [comment]
Note that I moved this back to `Needs Review` from the `Punted to 6.9.1` status.  Depending on how complete the resulting PR(s) are in 6.9, I'll open a follow-up issue for 6.9.1 to ensure we follow through on the complete guidance from @joedolson in https://github.com/WordPress/gutenberg/issues/72594#issuecomment-3482873199.

**Comment by t-hamano** [comment]
> Needs to be done, could be a point release:
> 
> 4. To add the necessary tree navigation, the right arrow should open the tree item (same as Enter currently does), and left arrow should close it when the parent node is focused. Up/Down arrows should navigate between individual tree items. Home should go to the first comment; End should go to the last comment.

I have submitted a new issue regarding this: https://github.com/WordPress/gutenberg/issues/72958
